tensorboard/plugins/beholder/README.md

```diff
@@ -0,0 +1,153 @@
+# The Beholder Plugin
+
+The Beholder plugin shows a live video feed of tensor data in TensorBoard during
+model training. It can display model variable values, arbitrary NumPy arrays
+(e.g. for gradients or activations), or pre-existing image frames.
+
+For now, Beholder should be considered a **beta release**.
+
+Beholder was originally developed by [Chris Anderson][chrisranderson], who also
+created this [demo video for Beholder][demo-video].
+
+[chrisranderson]: https://github.com/chrisranderson
+[demo-video]: https://www.youtube.com/watch?v=06HjEr0OX5k
+
+![The Beholder Dashboard](docs/beholder.png)
+
+## Instrumenting a Model
+
+To use Beholder, import and instantiate the `Beholder` class in your model code
+and from the training loop call the `update` method with a `Session` argument:
+
+```python
+from tensorboard.plugins.beholder import Beholder
+beholder = Beholder(LOG_DIRECTORY)
+
+# inside train loop
+beholder.update(
+  session=sess,
+  arrays=list_of_np_ndarrays,  # optional argument
+  frame=np_ndarray,  # optional argument
+)
+```
+
+By default, Beholder will visualize `tf.trainable_variables()`, but two optional
+arguments can provide additional data sources for visualization:
+
+- `arrays`: a list of NumPy `ndarray` values, interpreted as described below
+
+- `frame`: an `ndarray` of shape `[height, width, channels]` for direct display
+  via [`tf.image.encode_png`][encode-png] (if omitted, `channels` is set to 1),
+  or a function that returns such an array (to skip generation if not displayed)
+
+[encode-png]: https://www.tensorflow.org/api_docs/python/tf/image/encode_png
+
+For code using `tf.train.MonitoredSession`, there's also `BeholderHook`:
+
+```python
+from tensorboard.plugins.beholder import BeholderHook
+beholder_hook = BeholderHook(LOG_DIRECTORY)
+with MonitoredSession(..., hooks=[beholder_hook]) as sess:
+  sess.run(train_op)
+```
+
+## The Beholder Dashboard
+
+Launch TensorBoard pointing at the same directory (e.g `--logdir LOG_DIRECTORY`)
+and select the "Beholder" tab.
+
+### Data Visualization
+
+Beholder's primary feature is visualizing arbitrary NumPy arrays of data in a
+single video stream. Within the stream, horizontal *sections* represent each
```

Comment:
This is a great breakdown of the vocabulary.
nit: I personally find italics to be barely perceivable for sans serif fonts ... maybe bold instead.

Reply:
Thanks!  Hmm, bold seems too attention-getting to me when the point is just to show we're defining terminology - how about if I quote the terms instead?  PTAL.

Reply:
Hmm, yeah, that looks fine to me.